### ISSUE TITLE
failed to run fennel

### ISSUE THREAD

**Issue by wsdjeg** [bakpakin/Fennel#351]
I just download the fennel from: https://fennel-lang.org/downloads/fennel-0.9.2-windows32.exe, and move it into `D:\bin`, rename it to `fennel.exe`, and run `fennel` in `cmd.exe`, I got:

```
[string "src.launcher"]:1: module 'fennel' not found:
        no field package.preload['fennel']
        no file 'D:\bin\lua\fennel.lua'
        no file 'D:\bin\lua\fennel\init.lua'
        no file 'D:\bin\fennel.lua'
        no file 'D:\bin\fennel\init.lua'
        no file 'D:\bin\..\share\lua\5.3\fennel.lua'
        no file 'D:\bin\..\share\lua\5.3\fennel\init.lua'
        no file '.\fennel.lua'
        no file '.\fennel\init.lua'
        no file 'D:\bin\fennel.dll'
        no file 'D:\bin\..\lib\lua\5.3\fennel.dll'
        no file 'D:\bin\loadall.dll'
        no file '.\fennel.dll'
stack traceback:
        [C]: in function 'require'
        [string "src.launcher"]:1: in main chunk
        (...tail calls...)

```

**Comment by technomancy** [comment]
Sorry; I don't have a windows machine so there's not much I can do to test/debug this kind of thing. Open to suggestions for possible fixes.

**Comment by alexjgriffith** [comment]
It looks like fennel.lua is not baked into the exe, or it cannot access itself in the exe (I am not sure how they build it). 

To fix this for now:
1. Download the fennel-0.9.2 lua build output from [https://fennel-lang.org/downloads/fennel-0.9.2](https://fennel-lang.org/downloads/fennel-0.9.2)
2. remove the first line from the file (`#!/usr/bin/env lua`)
3. rename the file `fennel.lua`
4. Make sure the file is saved in the same directory as fennel.exe (in your case `D:\bin`). You could alternatively stick it in any of the directories listed in your error message. For example, `D:\bin\..\share\lua\5.3\fennel.lua`

**Comment by technomancy** [comment]
It sounds like that means `--compile-binary` isn't propagating `--require-as-include` somehow when compiling `fennel.exe` using `i686-w64-mingw32-gcc`.

Does `--compile-binary` work on Windows for normal Fennel projects? Does `--require-as-include` work on its own?

**Comment by alexjgriffith** [comment]
There is some real odd behavior going on

`--require-as-include` works fine on windows build for 9.2 for a simple test case

``` lua
griffita@rig:~/love/okastan/src$ ~/github/fennel.exe --compile test.fnl 
local inc = require("include")
return nil
griffita@rig:~/love/okastan/src$ ~/github/fennel.exe --compile --require-as-include test.fnl 
local inc
package.preload["include"] = package.preload["include"] or function(...)
  return {example = true}
end
inc = require("include")
return nil
```

It does not work at all for src/launcher.fnl :/

``` lua
griffita@rig:~/github/Fennel$ ../fennel.exe --compile --require-as-include src/launcher.fnl | head -2
local fennel = require("fennel")
local unpack = (table.unpack or _G.unpack)
griffita@rig:~/github/Fennel$ ../fennel.exe --compile  src/launcher.fnl | head -2
local fennel = require("fennel")
local unpack = (table.unpack or _G.unpack)
```

To make matters more confusing,

When I run `--compile-binary` with fennel V 9.1 (built myself last week), `x86_64-w64-mingw32-gcc` and the `luajit` library, it compiles without an issue. The binary can be moved out of its compilation environment and it continues to work. 

But then if i try again using the official windows builds it doesn't work.

No idea what is happening...

**Comment by technomancy** [comment]
I think I found the problem here; there was a missing `--add-fennel-path` argument in the Makefile for the Windows build. Not sure why this wasn't affecting 0.9.1 though!

I've updated the downloads on https://fennel-lang.org/setup so please let me know if this fixes the problem!

Fixed in d8a3192

**Comment by wsdjeg** [comment]
thanks
